code/modules/religion/rites.dm

```diff
@@ -411,3 +411,98 @@
+	var/turf/altar_turf = get_turf(religious_tool)
+	new /obj/effect/temp_visual/bluespace_fissure/long(altar_turf)
+	user.visible_message("<span class'notice'>A tear in reality appears above the altar!</span>")
+	var/list/jobbans = list(ROLE_BRAINWASHED, ROLE_DEATHSQUAD, ROLE_DRONE, ROLE_LAVALAND, ROLE_MIND_TRANSFER, ROLE_POSIBRAIN, ROLE_SENTIENCE)
+	var/list/candidates = pollGhostCandidates("Do you wish to be summoned as a Holy Carp?", jobbans, null, FALSE)
+	if(!length(candidates))
+		new /obj/effect/gibspawner/generic(altar_turf)
+		user.visible_message("<span class='warning'>The carp pool was not strong enough to bring forth a space carp.")
+		GLOB.religious_sect?.adjust_favor(400, user)
+		return NOT_ENOUGH_PLAYERS
+	var/mob/dead/observer/selected = pick_n_take(candidates)
+	var/datum/mind/M = new /datum/mind(selected.key)
+	var/carp_species = pick(/mob/living/simple_animal/hostile/carp/megacarp, /mob/living/simple_animal/hostile/carp)
+	var/mob/living/simple_animal/hostile/carp = new carp_species(altar_turf)
+	carp.name = "Holy Space-Carp ([rand(1,999)])"
+	carp.key = selected.key
+	carp.sentience_act()
+	carp.maxHealth += 100
+	carp.health += 100
+	M.transfer_to(carp)
+	if(GLOB.religion)
+		carp.mind?.holy_role = HOLY_ROLE_PRIEST
+		to_chat(carp, "There is already an established religion onboard the station. You are an acolyte of [GLOB.deity]. Defer to the Chaplain.")
+		GLOB.religious_sect?.on_conversion(carp)
+	playsound(altar_turf, 'sound/effects/slosh.ogg', 50, TRUE)
+	return ..()
+
+/datum/religion_rites/summon_carpsuit
+	name = "Summon Carp-Suit"
+	desc = "Summons a Space-Carp Suit"
+	ritual_length = 60 SECONDS
+	ritual_invocations = list("We shall become one ...",
+	"... we shall blend in ...",
+	"... we shall join in the ways of the carp ...",
+	"... grant us new clothing ...")
+	invoke_msg = "So we can swim."
+	favor_cost = 300
+	var/obj/item/clothing/suit/chosen_clothing
+
+/datum/religion_rites/summon_carpsuit/perform_rite(mob/living/user, atom/religious_tool)
+	var/turf/T = get_turf(religious_tool)
+	var/list/L = T.contents
+	if(!locate(/obj/item/clothing/suit) in L)
+		to_chat(user, "<span class='warning'>There is no suit clothing on the altar!</span>")
+		return FALSE
+	for(var/obj/item/clothing/suit/apparel in L)
+		chosen_clothing = apparel //the apparel has been chosen by our lord and savior
+		return ..()
+	return FALSE
+
+/datum/religion_rites/summon_carpsuit/invoke_effect(mob/living/user, atom/religious_tool)
+	if(!QDELETED(chosen_clothing) && get_turf(religious_tool) == chosen_clothing.loc) //check if the same clothing is still there
+		user.visible_message("<span class'notice'>The [chosen_clothing] transforms!</span>")
+		QDEL_NULL(chosen_clothing)
+		new /obj/item/clothing/suit/space/hardsuit/carp/old(get_turf(religious_tool))
+		playsound(get_turf(religious_tool), 'sound/effects/slosh.ogg', 50, TRUE)
+		chosen_clothing = null //our lord and savior no longer cares about this apparel
```

Comment:
this null call is still here

Reply:
Also, this doesn't address the fact that qdeleting the clothing will remove its contents.

The contents should be dumped before qdeletion.

Reply:
All fixed, calling obj.destruction instead, this also removes the need for qdel but will require the null.